src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -527,9 +1430,52 @@ private void updateLineShapePoints(final Line line)
             for (final Point point : this.rawAtlas.pointsAt(location))
             {
                 this.statistics.recordProcessedPoint();
-                this.afterSlicingLinesAndPoints.updatePointTags(point.getIdentifier(),
+                this.slicedPointAndLineChanges.updatePointTags(point.getIdentifier(),
                         createPointTags(location, true));
             }
         }
     }
+
+    /**
+     * Updates {@link Relation} tags with the {@link SyntheticRelationMemberAdded} value to keep
+     * track of any synthetic line members that were created and added to the relation.
+     *
+     * @param relationIdentifier
+     *            The {@link Relation} identifier whose tags to update
+     * @param newLineIdentifier
+     *            The {@link Line} identifier that got created
+     */
+    private void updateSyntheticRelationMemberTag(final long relationIdentifier,
+            final long newLineIdentifier)
+    {
+        final Map<String, String> updatedTags = this.slicedRelationChanges.getUpdatedRelationTags()
+                .get(relationIdentifier);
+        if (updatedTags == null)
+        {
+            // No updated tags exist for this relation, create new ones and insert synthetic value
+            final Map<String, String> newTags = new HashMap<>();
+            newTags.put(SyntheticRelationMemberAdded.KEY.toString(),
+                    String.valueOf(newLineIdentifier));
+            this.slicedRelationChanges.updateRelationTags(relationIdentifier, newTags);
+        }
+        else
+        {
+            // Updated tags exist
+            if (updatedTags.containsKey(SyntheticRelationMemberAdded.KEY))
+            {
+                // Synthetic key exists, need to append another member
+                final String newValue = updatedTags.get(SyntheticRelationMemberAdded.KEY)
+                        + SyntheticRelationMemberAdded.MEMBER_DELIMITER
+                        + String.valueOf(newLineIdentifier);
+                updatedTags.put(SyntheticRelationMemberAdded.KEY.toString(), newValue);
+            }
+            else
+            {
+                // Synthetic key doesn't exist, need to insert a new value
+                updatedTags.put(SyntheticRelationMemberAdded.KEY.toString(),
+                        String.valueOf(newLineIdentifier));
+            }
+            this.slicedRelationChanges.updateRelationTags(relationIdentifier, updatedTags);
+        }
+    }
 }
```

Comment:
This file is now 1500 lines long, which is getting to the point of considering why it is so long and possibly if there are any changes that can be made to clean it up or break it up. Readability, testability and understanding classes of this size start breaking down fairly quickly once you start getting up to that many lines.

Reply:
I totally agree with you! See line 1185 for a TO-DO calling out this exact item. I'm working on adding a few more test cases and then separating this behavior into something more concise.

Reply:
Nice, I obviously skipped over the TODO.